$NetBSD: patch-configure.in,v 1.3 2016/04/26 22:51:30 schnoebe Exp $

convert configure to use more configure paths instead of hard coded
paths.

--- configure.in.orig	2016-03-16 10:38:34.000000000 +0000
+++ configure.in
@@ -34,14 +34,15 @@ AC_DISABLE_STATIC
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
+AM_PROG_CC_C_O
 AC_PROG_INSTALL
+AC_LIBTOOL_DLOPEN
 AC_PROG_LIBTOOL
 
 # Checks for required libraries, don't set global -lpthread, -lm, -ljpeg, ... here, set in Makefile.
 AC_CHECK_LIB([pthread], [pthread_create], [LIBS="$LIBS"], [AC_MSG_ERROR([cannot find libpthread support], 7)])
 AC_CHECK_LIB([m], [pow], [LIBS="$LIBS"], [AC_MSG_ERROR([cannot find libm math support], 10)])
 AC_CHECK_LIB([jpeg], [jpeg_set_defaults],[LIBS="$LIBS"], [AC_MSG_ERROR(["cannot find libjpeg support"], 102)])
-AC_CHECK_LIB([dl], [dlopen], [LIBS="$LIBS" AC_DEFINE([HAVE_LIBDL])], [AC_MSG_ERROR(["cannot find libdl support"], 104)])
 
 # Checks for required header files.
 AC_CHECK_HEADERS(pthread.h,, [AC_MSG_ERROR([cannot find pthread-devel support], 7)])
@@ -267,7 +268,7 @@ AC_ARG_ENABLE(apparmor_build,
   [  --enable-apparmor_build    enable apparmor build (default=no)],
   apparmor_build=$enableval, apparmor_build=no)
 if test "$apparmor_build" = "yes"; then
-   if test -d /etc/apparmor.d/; then
+   if test -d $sysconfdir/apparmor.d/; then
       AC_MSG_RESULT(yes)
    else
       AC_MSG_ERROR([cannot find apparmor support])
@@ -333,37 +334,37 @@ else
 fi
 AM_CONDITIONAL(LIBUSB01_BUILD, test x$libusb01_build = xyes)
 
-AC_ARG_WITH(cupsbackenddir, AC_HELP_STRING([--with-cupsbackenddir=DIR], [set cups backend install directory [default=/usr/lib/cups/backend]]),
-   cupsbackenddir=$withval, cupsbackenddir="/usr/lib/cups/backend")
+AC_ARG_WITH(cupsbackenddir, AC_HELP_STRING([--with-cupsbackenddir=DIR], [set cups backend install directory [default=libdir/lib/cups/backend]]),
+   cupsbackenddir=$withval, cupsbackenddir="$libdir/lib/cups/backend")
 
-AC_ARG_WITH(cupsfilterdir, AC_HELP_STRING([--with-cupsfilterdir=DIR], [set cups filter install directory [default=/usr/lib/cups/filter]]),
-   cupsfilterdir=$withval, cupsfilterdir="/usr/lib/cups/filter")
+AC_ARG_WITH(cupsfilterdir, AC_HELP_STRING([--with-cupsfilterdir=DIR], [set cups filter install directory [default=libdir/lib/cups/filter]]),
+   cupsfilterdir=$withval, cupsfilterdir="$libdir/lib/cups/filter")
 
-AC_ARG_WITH(icondir, AC_HELP_STRING([--with-icondir=DIR], [set hplip.desktop install directory [default=/usr/share/applications]]),
-   icondir=$withval, icondir="/usr/share/applications")
+AC_ARG_WITH(icondir, AC_HELP_STRING([--with-icondir=DIR], [set hplip.desktop install directory [default=datadir/applications]]),
+   icondir=$withval, icondir="$datadir/applications")
 
-AC_ARG_WITH(systraydir, AC_HELP_STRING([--with-systraydir=DIR], [set hplip-systray.desktop install directory [default=/etc/xdg/autostart]]),
-   systraydir=$withval, systraydir="/etc/xdg/autostart")
+AC_ARG_WITH(systraydir, AC_HELP_STRING([--with-systraydir=DIR], [set hplip-systray.desktop install directory [default=sysconfdir/xdg/autostart]]),
+   systraydir=$withval, systraydir="$sysconfdir/xdg/autostart")
 
 AC_ARG_WITH(hpppddir, AC_HELP_STRING([--with-hpppddir=DIR], [set hp ppd install directory [default=datadir/ppd/HP]]),
    hpppddir=$withval, hpppddir="$datadir/ppd/HP")
    
 # AppArmor Changes
-AC_ARG_WITH(apparmordir, AC_HELP_STRING([--with-apparmordir=DIR], [set apparmor profile install directory [default=/etc/apparmor.d]]),
-   apparmordir=$withval, apparmordir="/etc/apparmor.d")
+AC_ARG_WITH(apparmordir, AC_HELP_STRING([--with-apparmordir=DIR], [set apparmor profile install directory [default=sysconfdir/apparmor.d]]),
+   apparmordir=$withval, apparmordir="$sysconfdir/apparmor.d")
 
-if test -f "/usr/share/cups/mime/mime.types"; then
-   abs_mimedir="/usr/share/cups/mime/"
+if test -f "$datadir/cups/mime/mime.types"; then
+   abs_mimedir="$datadir/cups/mime/"
 else
-   abs_mimedir="/etc/cups/"
+   abs_mimedir="$sysconfdir/cups/"
 fi
 AC_ARG_WITH(mimedir, AC_HELP_STRING([--with-mimedir=DIR], [set cups mime.types location [default=abs_mimedir]]),
    mimedir=$withval, mimedir="$abs_mimedir")
 
-AC_ARG_WITH(docdir, AC_HELP_STRING([--with-docdir=DIR], [set hplip documentation directory [default=datadir/doc]]),
+AC_ARG_WITH(docdir, AC_HELP_STRING([--with-docdir=DIR], [set hplip documentation directory [default=datadir/doc/hplip-VERSION]]),
    hpdocdir=$withval, hpdocdir="$datadir/doc/hplip-$VERSION")
 
-AC_ARG_WITH(htmldir, AC_HELP_STRING([--with-htmldir=DIR], [set hplip html documentation directory [default=datadir/doc]]),
+AC_ARG_WITH(htmldir, AC_HELP_STRING([--with-htmldir=DIR], [set hplip html documentation directory [default=datadir/doc/hplip-VERSION]]),
    hphtmldir=$withval, hphtmldir="$datadir/doc/hplip-$VERSION")
 
 AC_MSG_CHECKING([for foomatic ppd install])
@@ -488,12 +489,12 @@ AM_CONDITIONAL(POLICYKIT_INSTALL, test x
 
 # Determine where the PolicyKit policy file goes
 if test "$policykit" = "yes"; then
-   if test -d /usr/share/polkit-1; then
-      policykit_dir=/usr/share/polkit-1/actions
-   elif test -d /usr/share/PolicyKit; then
-      policykit_dir=/usr/share/PolicyKit/policy
+   if test -d "$datadir/polkit-1"; then
+      policykit_dir="$datadir/polkit-1/actions"
+   elif test -d "$datadir/PolicyKit"; then
+      policykit_dir="$datadir/PolicyKit/policy"
    else
-      policykit_dir=/usr/share/PolicyKit/policy
+      policykit_dir="$datadir/PolicyKit/policy"
    fi
 else
    policykit_dir=
@@ -602,6 +603,7 @@ AC_DEFINE_DIR([abs_sbindir], [sbindir])
 AC_DEFINE_DIR([abs_hpppddir], [hpppddir])
 AC_DEFINE_DIR([abs_docdir], [hpdocdir])
 AC_DEFINE_DIR([abs_htmldir], [hphtmldir])
+AC_DEFINE_DIR([abs_icondir], [icondir])
 abs_ppddir=${abs_hpppddir%/*}
 AC_DEFINE_DIR([abs_drvdir], [drvdir])
 
